lib/src/widgets/board.dart

```diff
@@ -0,0 +1,146 @@
+  final IMap<Square, Annotation>? annotations;
+  final GlobalKey? boardKey;
+  final Setup setup;
+
+  @override
+  Widget build(BuildContext context) {
+    final Map<Square, Square> castlingMap = {
+      Square.a1: Square.c1,
+      Square.a8: Square.c8,
+      Square.h1: Square.g1,
+      Square.h8: Square.g8,
+    };
+
+    GameData? modifiedGameData;
+
+    MapEntry<Square, ISet<Square>> mapper(Square sq, ISet<Square> moves) {
+      return MapEntry(
+        sq,
+        setup.board.kings.squares.contains(sq) && //king move
+                setup.castlingRights.squares.intersectsWith(
+                  moves, //king can castle
+                )
+            ? (switch (boardPrefs.castlingMethod) {
+              CastlingMethod.kingOverRook => moves.removeAll(castlingMap.values),
+              CastlingMethod.kingTwoSquares => moves.removeAll(castlingMap.keys),
+              _ => moves,
+            })
+            : moves,
+      );
+    }
+
+    if (gameData != null) {
+      modifiedGameData = GameData(
+        playerSide: gameData!.playerSide,
+        sideToMove: gameData!.sideToMove,
+        validMoves: gameData!.validMoves.map(mapper),
+        promotionMove: gameData!.promotionMove,
+        onMove: gameData!.onMove,
+        onPromotionSelection: gameData!.onPromotionSelection,
+        premovable: gameData?.premovable,
+        isCheck: gameData?.isCheck,
+      );
+    }
+
+    final board = Chessboard(
+      key: boardKey,
+      size: size,
+      fen: fen,
+      orientation: orientation,
+      game: modifiedGameData ?? gameData,
+      lastMove: lastMove,
+      shapes: shapes,
+      settings: settings,
+      annotations: annotations,
+    );
+
+    if (boardOverlay != null) {
+      return SizedBox.square(
+        dimension: size,
+        child: Stack(
+          children: [
+            board,
+            SizedBox.square(
+              dimension: size,
+              child: Center(
+                child: SizedBox(
+                  width: (size / 8) * 6.6,
+                  height: (size / 8) * 4.6,
+                  child: boardOverlay,
+                ),
+              ),
+            ),
+          ],
+        ),
+      );
+    } else if (error != null) {
+      return SizedBox.square(
+        dimension: size,
+        child: Stack(children: [board, _ErrorWidget(errorMessage: error!, boardSize: size)]),
+      );
+    }
+
+    return board;
+  }
+}
+
+class _ErrorWidget extends StatelessWidget {
+  const _ErrorWidget({required this.errorMessage, required this.boardSize});
+  final double boardSize;
+  final String errorMessage;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox.square(
+      dimension: boardSize,
+      child: Center(
+        child: Padding(
+          padding: const EdgeInsets.all(16.0),
+          child: Container(
+            decoration: BoxDecoration(
+              color:
+                  Theme.of(context).platform == TargetPlatform.iOS
```

Comment:
Since the new theme we can remove these distinctions between iOS and android. So use `colorScheme.surface` on both platforms.

I should have changed it in the new theme PR, but let's do it here.